docs/source-pytorch/versioning.rst

```diff
@@ -49,19 +53,99 @@ Lightning's development is driven by research and best practices in a rapidly de
 
 For API removal, renaming or other forms of backwards-incompatible changes, the procedure is:
 
-#. A deprecation process is initiated at a minor version ``X``, producing a deprecation warning at runtime and in the documentation.
-#. The deprecated API remains unchanged during the deprecation phase for two minor versions.
-#. The breaking change takes effect at a minor version ``X+2``.
-#. From version ``X+2`` onward, the deprecation warning gets converted into a helpful error, which will remain as long as possible.
+#. A deprecation process is initiated at a minor version ``MAJOR.MINOR.PATCH``, producing a deprecation warning at runtime and removing it from the documentation.
+#. The deprecated API remains unchanged during the deprecation phase for two minor versions or the next major update, whichever comes first.
+#. The breaking change is done in version ``MAJOR.MINOR+2.0``, or ``MAJOR+1.0.0``, whichever comes first.
+#. From that version onward, the deprecation warning gets converted into a helpful error, which will remain until next major release.
```

Comment:
This is more difficult to understand. Can we consider this instead:

X.Y.0
(1.5.0)

and

X.(Y+2).0
(1.7.0)

to explain the situation?